include/sot/core/matrix-geometry.hh

```diff
@@ -34,6 +34,54 @@
 /* --------------------------------------------------------------------- */
 /* --------------------------------------------------------------------- */
 /* --------------------------------------------------------------------- */
+
+namespace Eigen
+{
+  #define EIGEN_MAKE_TYPEDEFS(Type, TypeSuffix, Size, SizeSuffix)   \
+  /** \ingroup matrixtypedefs */                                    \
+  typedef Matrix<Type, Size, Size> Matrix##SizeSuffix##TypeSuffix;  \
+  /** \ingroup matrixtypedefs */                                    \
+  typedef Matrix<Type, Size, 1>    Vector##SizeSuffix##TypeSuffix;  \
+  /** \ingroup matrixtypedefs */                                    \
+  typedef Matrix<Type, 1, Size>    RowVector##SizeSuffix##TypeSuffix;
+
+  #define EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, Size)         \
+  /** \ingroup matrixtypedefs */                                    \
+  typedef Matrix<Type, Size, Dynamic> Matrix##Size##X##TypeSuffix;  \
+  /** \ingroup matrixtypedefs */                                    \
+  typedef Matrix<Type, Dynamic, Size> Matrix##X##Size##TypeSuffix;
+
+  #define EIGEN_MAKE_TYPEDEFS_ALL_SIZES(Type, TypeSuffix) \
+  EIGEN_MAKE_TYPEDEFS(Type, TypeSuffix, 1, 1) \
+  EIGEN_MAKE_TYPEDEFS(Type, TypeSuffix, 5, 5) \
+  EIGEN_MAKE_TYPEDEFS(Type, TypeSuffix, 6, 6) \
+  EIGEN_MAKE_TYPEDEFS(Type, TypeSuffix, 7, 7) \
+  EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 1) \
+  EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 5) \
+  EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 6) \
+  EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 7)
+
+  EIGEN_MAKE_TYPEDEFS_ALL_SIZES(int,                  i)
+  EIGEN_MAKE_TYPEDEFS_ALL_SIZES(float,                f)
+  EIGEN_MAKE_TYPEDEFS_ALL_SIZES(double,               d)
+  EIGEN_MAKE_TYPEDEFS_ALL_SIZES(std::complex<float>,  cf)
+  EIGEN_MAKE_TYPEDEFS_ALL_SIZES(std::complex<double>, cd)
```

Comment:
These are side remarks:
- I might be wrong, but I doubt these typedefs are used,
- I personally do not like to put thinks in other packages namespace, when avoidable.

Reply:
Thanks Joseph for your careful review. I do agree with both remarks.